app/models/monthly_service_metrics.rb

```diff
@@ -26,6 +27,35 @@ def calls_received_metric
     end
   end
 
+  def number_of_transactions
+    if !transactions_processed_with_intended_outcome.nil?
```

Comment:
To test whether you should be checking a particular metric or not, you can use the service object and check the attribute ```transactions_processed_with_intended_outcome_applicable```

Reply:
It may be easier to merge https://github.com/alphagov/service-performance/pull/114 first so you could 
```
if service.metric_application(:transactions_processed_with_indended_outcome) && ....
```